app/Commands/GenerateTrackFilesCommand.php

```diff
@@ -180,7 +204,8 @@ private function isLosslessFile($file) {
      * @param string $format
      * @return TrackFile|null
      */
-    private function trackFileExists(string $format) {
-        return $this->track->trackFiles()->where('format', $format)->first();
+    private function trackFileExists(string $format)
+    {
+        return $this->track->trackFilesForAllVersions()->where('format', $format)->where('version', $this->version)->first();
```

Comment:
Several issues here:

- This code will always execute a database query, even if the `trackFilesForAllVersions` relation is already loaded. Was that intentional?

- The function lacks a return type declaration. I realize this is a legacy function but this would be a good time to add a type declaration.